Framework/Algorithms/src/RemoveBackground.cpp

```diff
@@ -123,11 +127,11 @@ void RemoveBackground::exec() {
   for (int hist = 0; hist < histnumber; ++hist) {
     PARALLEL_START_INTERUPT_REGION
     // get references to output Workspace X-arrays.
-    MantidVec &XValues = outputWS->dataX(hist);
+    auto &XValues = outputWS->mutableX(hist);
     // get references to output workspace data and error. If it is new
     // workspace, data will be copied there, if old, modified in-place
-    MantidVec &YValues = outputWS->dataY(hist);
-    MantidVec &YErrors = outputWS->dataE(hist);
+    auto &YValues = outputWS->mutableY(hist);
+    auto &YErrors = outputWS->mutableE(hist);
```

Comment:
Can we pass a `Histogram` to `removeBackground`? It looks inefficient at first sight, because it would not be an in-place modification (`outputWS->setHistogram(hist, removeBackground(outputWS->histogram(hist))`), but `outputWS` is just created above and should have all shared data, so there would be basically no overhead. Except on second sight, if it is an in-place modification of the input workspace...

What do you think?